src/mvBasicWidgets.cpp

```diff
@@ -6398,17 +6431,17 @@ DearPyGui::draw_tooltip(ImDrawList* drawlist, mvAppItem& item)
 			}
 			apply_local_theming(&item);
 
-			ImGui::BeginTooltip();
+			if(ImGui::BeginTooltip()) {
```

Comment:
While this condition might makes tooltip rendering more efficient in some cases (not in the current version of ImGui though :grin:), it will also lead to item state "getting stuck" like it was on menu widgets in #2245. We need to update the state even when `BeginTooltip` returns `false`. Obviously `state.visible` will be false in this case; `rectSize` should probably be (0, 0); not sure about content region - probably (0, 0) too.

Reply:
I applied this, tell me if it's correct in the next commit. Otherwise, if this can lead to others issues, I can simply remove that if statement

Reply:
Looks good, but it would be more robust to update `item.state.lastFrameUpdate` in both branches (you can just move it out of the `if` block, better place _after_ if/else). `get_item_state` and handlers use that field to validate the state (it just happens that in this particular case forgetting to update it doesn't lead to any visible effects :) - an invalid state here looks just like the valid one).